drivers/ethernet/eth_native_posix.c

```diff
@@ -356,6 +363,35 @@ static void eth_iface_init(struct net_if *iface)
 	net_if_set_link_addr(iface, ll_addr->addr, ll_addr->len,
 			     NET_LINK_ETHERNET);
 
+#ifdef CONFIG_NET_IPV4
+	if (ipv4_addr_cmd_opt != NULL) {
+		if (net_addr_pton(AF_INET, ipv4_addr_cmd_opt, &addr)) {
+			NET_ERR("Invalid address: %s", ipv4_addr_cmd_opt);
+			return;
+		}
+
+		net_if_ipv4_addr_add(iface, &addr, NET_ADDR_MANUAL, 0);
+	}
+
+	if (ipv4_nm_cmd_opt != NULL) {
+		if (net_addr_pton(AF_INET, ipv4_nm_cmd_opt, &netmask)) {
+			NET_ERR("Invalid netmask: %s", ipv4_nm_cmd_opt);
+			return;
+		}
+
+		net_if_ipv4_set_netmask_by_addr(iface, &addr, &netmask);
```

Comment:
If the user made a mistake, and provided netmask param only, we'd be using uninitialized `addr` variable here. Perhaps it'd make sense to process `ipv4_nm_cmd_opt` only if the user provided address as well?